### PR TITLE
[OCPCLOUD-960] Add support for dedicated instance tenancy

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR adds support for dedicated instance tenancy.

### CODE REVIEW

File: pkg/apis/awsprovider/v1beta1/awsproviderconfig_types.go [hunk 1]
```diff
@@ -88,6 +88,9 @@ type AWSMachineProviderConfig struct {
 
 	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
 	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// Tenancy indicates if instance should run on shared or single-tenant hardware.
+	Tenancy string `json:"tenancy,omitempty"`
```
Comment (JoelSpeed) [inline pkg/apis/awsprovider/v1beta1/awsproviderconfig_types.go:93]: Might be worth setting this as a typed string and having the valid options be part of this package as exported variables? WDYT? 

I would also suggest adding a comment to this to explain the valid and default options

File: pkg/actuators/machine/instances.go [hunk 2]
```diff
@@ -342,6 +351,24 @@ func launchInstance(machine *machinev1.Machine, machineProviderConfig *awsprovid
...
+				dedicatedTenancy,
+				hostTenancy)
+		}
+
+		if placement == nil {
+			placement = &ec2.Placement{}
+		}
+
+		placement.Tenancy = &machineProviderConfig.Tenancy
+	}
```
Comment (JoelSpeed) [inline pkg/actuators/machine/instances.go:370]: Would this be simpler as a switch?

```suggestion
	switch instanceTenancy {
		case "":
			// Do nothing when not set
		case defaultTenancy, dedicatedTenancy, hostTenancy:
			if placement == nil {
				placement = &ec2.Placement{}
			}
			placement.Tenancy = &machineProviderConfig.Tenancy
		default:
			return nil, mapierrors.CreateMachine("invalid instance tenancy: %s. Allowed options are: %s,%s,%s",
				instanceTenancy,
				defaultTenancy,
				dedicatedTenancy,
				hostTenancy)
	}
```

**Review by JoelSpeed — COMMENTED** [review]
Looks good, left a couple of suggestions for possible improvements.

Do you plan to have a follow up PR that validates the options in the validating webhook for AWS? This is a very viable field for validating there IMO

Have you done any manual testing for this feature yet? Do we need any extra config on the AWS account to be able to do this?

**Review by JoelSpeed — COMMENTED** [review]
/approve

**Comment by JoelSpeed** [comment]
/retitle [OCPCLOUD-960] Add support for dedicated instance tenancy

**Review by michaelgugino — APPROVED** [review]
LGMT

**Review by elmiko — APPROVED** [review]
/lgtm

**Review by michaelgugino — APPROVED** [review]
/lgtm

**Pull request merged**